### PR TITLE
WT-5529 Improve usage of upd in txn.i

### PR DESCRIPTION
**Pull request by jeremytay90**
as per https://github.com/wiredtiger/wiredtiger/pull/5216#discussion_r374984109

### CODE REVIEW

File: src/include/txn.i [hunk 1]
```diff
@@ -856,21 +856,21 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT
...
     if (__wt_txn_visible(session, start.txnid, start.timestamp)) {
-        WT_RET(__wt_update_alloc(session, &buf, &upd, &size, WT_UPDATE_STANDARD));
-        upd->txnid = start.txnid;
-        upd->start_ts = upd->durable_ts = start.timestamp;
-        F_SET(upd, WT_UPDATE_RESTORED_FROM_DISK);
+        WT_RET(__wt_update_alloc(session, &buf, updp, &size, WT_UPDATE_STANDARD));
+        (*updp)->txnid = start.txnid;
+        (*updp)->start_ts = start.timestamp;
+        F_SET((*updp), WT_UPDATE_RESTORED_FROM_DISK);
         *updp = upd;
```
Comment (sueloverso) [inline src/include/txn.i]: Won't this line overwrite everything you just set?

Comment (jeremytay90) [reply src/include/txn.i]: Good point - I should've noticed!

File: src/include/txn.i [hunk 2]
```diff
@@ -856,21 +854,20 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT
...
+        WT_RET(__wt_update_alloc(session, &buf, updp, &size, WT_UPDATE_STANDARD));
+        (*updp)->txnid = start.txnid;
+        (*updp)->start_ts = start.timestamp;
+        F_SET((*updp), WT_UPDATE_RESTORED_FROM_DISK);
         return (0);
     }
 
     /* If there's no visible update in the update chain or ondisk, check the history store file. */
-    WT_ASSERT(session, upd == NULL);
+    WT_ASSERT(session, (*updp) == NULL);
```
Comment (keithbostic) [inline src/include/txn.i]: AFAICT, this assert is literally impossible, so there's no much point in keeping it.

File: src/include/txn.i [hunk 3]
```diff
@@ -856,21 +854,20 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT
...
     if (F_ISSET(S2C(session), WT_CONN_HS_OPEN) && !F_ISSET(S2BT(session), WT_BTREE_HS))
-        WT_RET_NOTFOUND_OK(__wt_find_hs_upd(session, cbt, &upd, false, &buf));
+        WT_RET_NOTFOUND_OK(__wt_find_hs_upd(session, cbt, updp, false, &buf));
 
     /* There is no BIRTHMARK in the history store file. */
-    WT_ASSERT(session, upd == NULL || upd->type != WT_UPDATE_BIRTHMARK);
+    WT_ASSERT(session, (*updp) == NULL || (*updp)->type != WT_UPDATE_BIRTHMARK);
```
Comment (keithbostic) [inline src/include/txn.i]: This assert should be part of the history store lookup, logically it makes no sense unless we're doing history store work.
I'd suggest:
```
*** txn.i	Fri Feb  7 00:05:07 2020
--- -	Fri Feb  7 00:09:11 2020
***************
*** 862,873 ****
      }
  
      /* If there's no visible update in the update chain or ondisk, check the history store file. */
!     WT_ASSERT(session, (*updp) == NULL);
!     if (F_ISSET(S2C(session), WT_CONN_HS_OPEN) && !F_ISSET(S2BT(session), WT_BTREE_HS))
          WT_RET_NOTFOUND_OK(__wt_find_hs_upd(session, cbt, updp, false, &buf));
  
!     /* There is no BIRTHMARK in the history store file. */
!     WT_ASSERT(session, (*updp) == NULL || (*updp)->type != WT_UPDATE_BIRTHMARK);
  
      /*
       * If we checked the update list, the ondisk value and the history store, we should return a
--- 862,873 ----
      }
  
      /* If there's no visible update in the update chain or ondisk, check the history store file. */
!     if (F_ISSET(S2C(session), WT_CONN_HS_OPEN) && !F_ISSET(S2BT(session), WT_BTREE_HS)) {
          WT_RET_NOTFOUND_OK(__wt_find_hs_upd(session, cbt, updp, false, &buf));
  
! 	/* There is no BIRTHMARK in the history store file. */
! 	WT_ASSERT(session, (*updp) == NULL || (*updp)->type != WT_UPDATE_BIRTHMARK);
!     }
  
      /*
       * If we checked the update list, the ondisk value and the history store, we should return a
```

File: src/include/txn.i (hunk 2 above)
Comment (jeremytay90) [reply src/include/txn.i]: Agreed - looks like if the converse is true, we are returning.

**Pull request merged**